### PR TITLE
216 hold note end time

### LINKED ISSUE #216: Change HoldNote endtime in notes timeline
As a story editor, I want to change the duration of the hold note so that I can change the duration without having to recreate the hold note.

* Click and drag the end of the hold note in the notes timeline to change the duration

### PR DESCRIPTION
**Pull request by Xenocidel**
Closes #216 
Cursor to change to <> when hovering over end of HoldNote in NotesTimeline to be done in #126

### CODE REVIEW

File: S2VX.Game/Story/Note/EditorHoldNote.cs [hunk 1]
```diff
@@ -29,6 +29,15 @@ public class EditorHoldNote : HoldNote {
             HitSoundTimes = new List<double>() { HitTime, EndTime };
         }
 
+        public override void UpdateEndTime(double endTime) {
+            if (endTime < HitTime) {
+                return;
+            }
+            EndTime = endTime;
+            ((HoldApproach)Approach).EndTime = EndTime;
```
Comment (maxrchung) [inline S2VX.Game/Story/Note/EditorHoldNote.cs:37]: Can you see if you can do an override of the Approach property for hold notes? This way we don't have to cast it like this. I think you can set Approach to virtual inside of S2VXNote.cs and then override it in HoldNote.cs. For more details: https://stackoverflow.com/a/9238902/13183186

Comment (Xenocidel) [reply S2VX.Game/Story/Note/EditorHoldNote.cs]: Is this good?

Comment (maxrchung) [reply S2VX.Game/Story/Note/EditorHoldNote.cs]: I think this is okay. Feels kind of odd having 2 properties point to the same thing. I was hoping that we'd be able to somehow elegantly override the same property as a HoldApproach, but don't think that's possible.

**Review by maxrchung — APPROVED** [review]

File: S2VX.Game/Story/Note/EditorHoldNote.cs (hunk 1 above)
Comment (Xenocidel) [reply S2VX.Game/Story/Note/EditorHoldNote.cs]: Yeah, not possible to override the same property name with a different type.

**Review by Naranja-Sagged — APPROVED** [review]

**Pull request merged**